### PR TITLE
Switch to native pytest in tests/utils

### PR DESCRIPTION
**Pull request by pquentin**
There are two different commits, which should help with the review.

### CODE REVIEW

File: tests/utils/convert_test.py [hunk 1]
```diff
@@ -17,23 +17,25 @@
...
+            assert convert.to_bool(value) is False
 
     def test_cannot_convert_invalid_value(self):
         values = ["Invalid", None, []]
         for value in values:
-            with self.assertRaises(ValueError, msg="Expect [%s] of type [%s] to fail to be converted." % (str(value), type(value))) as ctx:
+            with pytest.raises(ValueError) as ctx:
                 convert.to_bool(value)
-            self.assertEqual("Cannot convert [%s] to bool." % value, ctx.exception.args[0])
+            assert "Cannot convert [%s] to bool." % value == ctx.exception.args[0]
```
Comment (danielmitterdorfer) [inline tests/utils/convert_test.py]: We could also switch to f-strings here?

File: tests/utils/io_test.py [hunk 2]
```diff
@@ -47,24 +47,24 @@ def mock_red_hat(path):
 
 class IoTests(TestCase):
     def test_normalize_path(self):
-        self.assertEqual("/already/a/normalized/path", io.normalize_path("/already/a/normalized/path"))
-        self.assertEqual("/not/normalized", io.normalize_path("/not/normalized/path/../"))
-        self.assertEqual(os.path.expanduser("~"), io.normalize_path("~/Documents/.."))
+        assert io.normalize_path("/already/a/normalized/path") == "/already/a/normalized/path"
+        assert io.normalize_path("/not/normalized/path/../") == "/not/normalized"
+        assert os.path.expanduser("~") == io.normalize_path("~/Documents/..")
```
Comment (danielmitterdorfer) [inline tests/utils/io_test.py:51]: Should we reverse the order here so the call to `io.normalize_path` comes first? Then this is consistent with the other assertions.

**Review by danielmitterdorfer — APPROVED** [review]
I left two minor comments. Please feel free to address them but no need for another review round in any case. LGTM

File: tests/utils/io_test.py (hunk 2 above)
Comment (pquentin) [reply tests/utils/io_test.py]: Nice catch, thanks

File: tests/utils/convert_test.py (hunk 1 above)
Comment (pquentin) [reply tests/utils/convert_test.py]: My intention is to this with pyupgrade at some point. Fixed here, though!

**Pull request merged**